source/components/temp_doeclim_component.cpp

```diff
@@ -0,0 +1,528 @@
+/* Hector -- A Simple Climate Model
+   Copyright (C) 2014-2015  Battelle Memorial Institute
+
+   Please see the accompanying file LICENSE.md for additional licensing
+   information.
+*/
+/*
+ *  temp_doeclim_component.cpp
+ *  hector
+ *
+ *  Created by Ben Vega-Westhoff on 11/1/16.
+ *
+ */
+
+#include <boost/lexical_cast.hpp>
+#include <cmath>
+#include <limits>
+#define _USE_MATH_DEFINES
+
+#include "components/temp_doeclim_component.hpp"
+#include "core/core.hpp"
+#include "h_util.hpp"
+#include "models/simpleNbox.hpp"
+#include "visitors/avisitor.hpp"
+
+namespace Hector {
+  
+using namespace std;
+
+
+//------------------------------------------------------------------------------
+/*! \brief Constructor
+ */
+TempDOECLIMComponent::TempDOECLIMComponent() : internal_Ftot(0.0), last_Ftot(0.0) {
+}
+
+//------------------------------------------------------------------------------
+/*! \brief Destructor
+ */
+TempDOECLIMComponent::~TempDOECLIMComponent(){
+}
+    
+//------------------------------------------------------------------------------
+// documentation is inherited
+string TempDOECLIMComponent::getComponentName() const {
+    const string name = TEMP_DOECLIM_COMPONENT_NAME;
+    
+    return name;
+}
+
+
+//------------------------------------------------------------------------------
+/*! \brief              Calculates inverse of x and stores in y
+ *  \param[in] x        Assume x is setup like x = [a,b,c,d] -> x = |a, b|
+ *                                                                  |c, d|
+ *  \param[in] y        Inverted 1-d matrix
+ *  \returns            void, inverse is stored in y
+ */
+void TempDOECLIMComponent::invert_1d_2x2_matrix(double * x, double * y) {
```

Comment:
Since we depend on [Boost](http://boost.org) anyway, why not use its `ublas` library?

Reply:
Any Boost inversion implementation I saw in my cursory google search looked a good deal more complex than this simple algorithm for 2x2 inversion. But I could be wrong!

Reply:
That's correct.  If you know the matrix is 2x2, you can invert it very simply using analytic formulae.  It's similar to how you could solve an equation using the Newton-Raphson method, but if you happen to know it's quadratic, it's simpler to use the quadratic formula.